great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -257,6 +257,13 @@ def __init__(
                     "Credentials or an engine are required for a SqlAlchemyExecutionEngine."
                 )
 
+        # these are two backends where temp_table_creation is not supported we set the default value to False.
+        if self.engine.dialect.name.lower() in [
+            "trino",
+            "awsathena",  # WKS 202201 - AWS Athena currently doesn't support temp_tables.
+        ]:
+            self._create_temp_table = False
+
```

Comment:
@Shinnnyshinshin Importantly, please see these lines below:
```
        create_temp_table: bool = batch_spec.get(
            "create_temp_table", self._create_temp_table
        ) and (
            self.engine.dialect.name.lower()
            not in [
                "trino",
            ]
        )
```
and decide whether or not we need to change them in any way in light of the work you are doing.  Thanks!